action/flow/event/flowevents.go

```diff
@@ -0,0 +1,64 @@
+package event
+
+import (
+	"time"
+)
+
+type Status string
+
+const (
+	CREATED   = "Created"
+	COMPLETED = "Completed"
+	CANCELLED = "Cancelled"
+	FAILED    = "Failed"
+	SCHEDULED = "Scheduled"
+	SKIPPED   = "Skipped"
+	STARTED   = "Started"
+	WAITING   = "Waiting"
+	UNKNOWN   = "Created"
+)
+
+const FLOW_EVENT_TYPE = "flowevent"
+const TASK_EVENT_TYPE = "taskevent"
+
+// FlowEvent provides access to flow instance execution details
+type FlowEvent interface {
+	// Returns flow name
+	FlowName() string
+	// Returns flow ID
+	FlowID() string
+	// In case of subflow, returns parent flow name
+	ParentFlowName() string
+	// In case of subflow, returns parent flow ID
+	ParentFlowID() string
+	// Returns event time
+	EventTime() time.Time
+	// Returns current flow status
+	FlowStatus() Status
+	// Returns output data for completed flow instance
+	FlowOutput() map[string]interface{}
+	// Returns error for failed flow instance
+	FlowError() error
+}
+
+// TaskEvent provides access to task instance execution details
+type TaskEvent interface {
+	// Returns flow name
+	FlowName() string
+	// Returns flow ID
+	FlowID() string
+	// Returns task name
+	TaskName() string
+	// Returns task type
+	TaskType() string
+	// Returns task status
+	TaskStatus() Status
+	// Returns event time
+	EventTime() time.Time
```

Comment:
I think Time() is fine here because the object is TaskEvent, so event.Time() is clear.

Reply:
@fm-tibco : Make sense. Updated PR